indico/modules/events/registration/clone.py

```diff
@@ -22,48 +22,101 @@
 from indico.modules.events.registration.models.forms import RegistrationForm
 from indico.modules.events.registration.models.form_fields import RegistrationFormFieldData
 from indico.modules.events.registration.models.items import RegistrationFormItem, RegistrationFormSection
+from indico.modules.events.registration.models.registrations import Registration, RegistrationData
 from indico.util.i18n import _
 
 from MaKaC.conference import EventCloner
 
 
 class RegistrationFormCloner(EventCloner):
+    def __init__(self, *args, **kwargs):
+        super(RegistrationFormCloner, self).__init__(*args, **kwargs)
+        self._field_data_map = {}
+        self._clone_with_registrations = False
+
     def get_options(self):
         if not is_feature_enabled(self.event, 'registration'):
             return {}
-        enabled = bool(self._find_registration_forms().count())
-        return {'form': (_('Registration forms'), enabled, False)}
-
-    def clone(self, new_event, options):
-        if 'form' not in options:
-            return
-        attrs = get_simple_column_attrs(RegistrationForm) - {'start_dt', 'end_dt'}
-        for old_form in self._find_registration_forms():
-            new_form = RegistrationForm(event_id=new_event.id, **{attr: getattr(old_form, attr) for attr in attrs})
-            self._clone_form_items(old_form, new_form)
-            db.session.add(new_form)
-            db.session.flush()
+        forms_cloner_enabled = bool(self._find_registration_forms().count())
+        regs_cloner_enabled = bool(self._find_registration_forms()
+                                   .filter(RegistrationForm.registrations.any(Registration.is_active))
+                                   .count())
+        selected = self.event.getType() != 'conference'
+        return {
+            'form': (_('Registration forms'), forms_cloner_enabled, selected),
```

Comment:
I'd never preselect this one. That way people only have to deselect one if they don't want to clone meeting participants